cas-server-support-oauth/src/main/java/org/jasig/cas/config/OAuthConfiguration.java

```diff
@@ -51,14 +73,35 @@
     public Config oauthSecConfig() {
         final CasClient oauthCasClient = new CasClient(this.casLoginUrl);
         oauthCasClient.setName(CAS_OAUTH_CLIENT);
+        oauthCasClient.setCallbackUrlResolver((url, context) -> {
```

Comment:
OK, attributes are definitely not released without the changes here. 

`AccessToken.Authentication.Principal` has no attributes available that were previously resolved by CAS, because the principal is created during validation for the incoming service that is `https://mmoayyed.unicon.net:8443/cas/oauth2.0/callbackAuthorize?client_name=CasOAuthClient`